### PR TITLE
Add feedback tool

### PR DESCRIPTION
**Pull request by laviniat1996**
This PR adds a survey that collects user feedback and adds it to a Goggle Sheet. 

It uses Google Scripts to manager to fetch the data and add it to the Google Sheet.
The survey itself is a React component. I used swizzle to edit the footer and make this component built in at the end of every doc's page.

image

It works like this:

- Thumbs up -> Adds a plus one to the thumbs up count
- Thumbs down -> A text feedback input opens up, the user can leave it empty or add some text -> Submit Feedback -> Adds plus one to the thumbs down count, and the feedback text, if added

Video showcasing how it works:

https://www.loom.com/share/32573a0c4a2543fea94928f5fc48c3ff

### CODE REVIEW

**Comment by yulin-dong** [comment]
Is the feedback sent to the same doc or different doc from this branch build https://feedback.avalanche-docs.pages.dev/ vs https://docs.avax.network/ after deployment?

**Comment by laviniat1996** [comment]
> Is the feedback sent to the same doc or different doc from this branch build https://feedback.avalanche-docs.pages.dev/ vs https://docs.avax.network/ after deployment?

It's the same doc, it just adds a different URL to the doc.

**Comment by yulin-dong** [comment]
> > Is the feedback sent to the same doc or different doc from this branch build https://feedback.avalanche-docs.pages.dev/ vs https://docs.avax.network/ after deployment?
> 
> It's the same doc, it just adds a different URL to the doc.

I see. Is that too much work to use different sheet on the same google sheet doc for different URL?

**Comment by laviniat1996** [comment]
> > > Is the feedback sent to the same doc or different doc from this branch build https://feedback.avalanche-docs.pages.dev/ vs https://docs.avax.network/ after deployment?
> > 
> > 
> > It's the same doc, it just adds a different URL to the doc.
> 
> I see. Is that too much work to use different sheet on the same google sheet doc for different URL?

I think it's possible, and have thought of that myself, but there are some issues with this approach. It will be harder to have an overall look, as each page will be on a different sheet. More prone to errors, as Google Scripts has to handle multiple sheets. I was thinking to reset this sheet monthly, and create a sheet where we store data for each month.

**Comment by yulin-dong** [comment]
> > > > Is the feedback sent to the same doc or different doc from this branch build https://feedback.avalanche-docs.pages.dev/ vs https://docs.avax.network/ after deployment?
> > > 
> > > 
> > > It's the same doc, it just adds a different URL to the doc.
> > 
> > 
> > I see. Is that too much work to use different sheet on the same google sheet doc for different URL?
> 
> I think it's possible, and have thought of that myself, but there are some issues with this approach. It will be harder to have an overall look, as each page will be on a different sheet. More prone to errors, as Google Scripts has to handle multiple sheets. I was thinking to reset this sheet monthly, and create a sheet where we store data for each month.

What I meant is that to organize Google sheet based on https://7ef747cc.avalanche-docs.pages.dev/, https://feedback.avalanche-docs.pages.dev/ or https://docs.avax.network/, not the each dev doc itself.

**Comment by laviniat1996** [comment]
> > > > > Is the feedback sent to the same doc or different doc from this branch build https://feedback.avalanche-docs.pages.dev/ vs https://docs.avax.network/ after deployment?
> > > > 
> > > > 
> > > > It's the same doc, it just adds a different URL to the doc.
> > > 
> > > 
> > > I see. Is that too much work to use different sheet on the same google sheet doc for different URL?
> > 
> > 
> > I think it's possible, and have thought of that myself, but there are some issues with this approach. It will be harder to have an overall look, as each page will be on a different sheet. More prone to errors, as Google Scripts has to handle multiple sheets. I was thinking to reset this sheet monthly, and create a sheet where we store data for each month.
> 
> What I meant is that to organize Google sheet based on https://7ef747cc.avalanche-docs.pages.dev/, https://feedback.avalanche-docs.pages.dev/ or https://docs.avax.network/, not the each dev doc itself.

Oh, got it. Well, when we launch this live, I'll delete all those test links.

File: src/components/DocsRating.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        console.error('Error submitting feedback:', error);
+        setErrorMessage("Failed to submit feedback. Please check your internet connection and try again.");
+        // Reset the feedback input to allow the user to try again
+        setShowFeedbackInput(true);
+      });
+      
+  };
+  
+  return (
+    <div className="docsRating">
```
[the remark this answers is not in the document]
Comment (laviniat1996) [reply src/components/DocsRating.js:81]: @yulin-dong Maybe we can merge it like this, and I'll think of a way to implement spanish as well

**Review by frichards — APPROVED** [review]
🌈 🦄 Approved!
Great job! I did have 1 nit-picking comment about the value of the feedback. But it is not critical to change.

File: src/components/DocsRating.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    })
+    .catch(error => {
+      console.error('Error sending analytics:', error);
+      setErrorMessage("There was a problem sending your feedback analytics. Please try again later.");
+    });
+  };
+  
+  // Handle user feedback (thumbs up/down)
+  const giveFeedback = (value) => {
+    if (value === 0) {
```
[the remark this answers is not in the document]
Comment (laviniat1996) [reply src/components/DocsRating.js]: Addressed e90f6bf525d2fbad92e4061ce1ea41413a19dd95

**Review by yulin-dong — APPROVED** [review]

**Review by meaghanfitzgerald — APPROVED** [review]

**Pull request merged**